fortitude/src/rules/correctness/conditionals.rs

```diff
@@ -0,0 +1,134 @@
+use crate::ast::FortitudeNode;
+use crate::settings::Settings;
+use crate::{AstRule, FromAstNode};
+use anyhow::{Context, Result};
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_source_file::SourceFile;
+use ruff_text_size::TextSize;
+use tree_sitter::Node;
+
+/// ## What does it do?
+/// Checks for misleading semicolons in `if` statements.
+///
+/// ## Why is this bad?
+/// The following code may appear to execute two statements only if the `if`
+/// condition is true, but in actuality the second statement will always be
+/// executed:
+///
+/// ```f90
+/// if (condition) print *, "Hello"; print *, "World"
+/// ```
+///
+/// It is equivalent to:
+///
+/// ```f90
+/// if (condition) then
+///    print *, "Hello"
+/// end if
+/// print *, "World"
+/// ```
+///
+/// When applying fixes, the if statement is converted to the second form and
+/// the semicolon is removed.
+#[derive(ViolationMetadata)]
+pub(crate) struct IfStatementSemicolon {}
+
+impl AlwaysFixableViolation for IfStatementSemicolon {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Semicolon following inline if-statement is misleading".to_string()
+    }
+
+    fn fix_title(&self) -> String {
+        "Convert to `if(...) then` statement".to_string()
+    }
+}
+impl AstRule for IfStatementSemicolon {
+    fn check(_settings: &Settings, node: &Node, src: &SourceFile) -> Option<Vec<Diagnostic>> {
+        // If this is an `if (...) then` construct, exit early
+        if node
+            .named_descendants()
+            .any(|n| n.kind() == "end_if_statement")
```

Comment:
Rather than iterating over all the descendants, it might be sufficient to check `child(2)`, which should be the unnamed node `then`?

Reply:
Thanks, corrected in https://github.com/PlasmaFAIR/fortitude/pull/426/commits/ba5bbd1154741dd9d626abc96311728eafda03cd